### DISCUSSION TITLE
How are people passing providing a vault password with this extension?

### DISCUSSION THREAD

**Discussion by zoredache** [tomaciazek/vscode-ansible#35] [General]
I have ansible configured with a vault password script, the script retrieves the vault password from the bitwarden vault using the `bw` cli.

    [defaults]
    vault_password_file = ./bw_vault_password.sh

In an interactive shell I run a command like this `export BW_SESSION=$( bw unlock --raw )` to unlock my bitwarden vault, and store the session details in the environment.  

I am running Vscode on Windows, and my ansible project is under WS.

The syntax highlighting looks nice for this extension, linting fails because I don't really seem to have any useful way to provide a vault password, or set the BW_SESSION environment variable that will be used be used by the processes this extension starts to syntax check and lint.

Anyway I am getting errors like this, and I am not sure how to approach solving it.

> [internal-error] Unexpected error code 1 from execution of: ansible-playbook --syntax-check playbooks/install_docker.yml
Description: This error can be caused by internal bugs but also by custom rules. Instead of just stopping linter we generate there errors and continue processing. This allows users to add this rule to warn list until the root cause is fixed.

**Comment by tomaciazek** [comment]
Please try running `ansible-playbook --syntax-check playbooks/install_docker.yml` command and see what the issue might be. Also, you might provide a custom activation script (there is a setting for that), in which you set the appropriate variable, as long as it doesn't need to be interactive. If it does, do the interactive part, save the results in a file and read that in the custom activation script.
